internal/datastore/crdb/migrations/zz_migration.0008_add_transaction_metadata_table.go

```diff
@@ -0,0 +1,75 @@
+package migrations
+
+import (
+	"context"
+	"strings"
+
+	"github.com/jackc/pgx/v5"
+)
+
+const (
+	// ttl_expiration_expression support was added in CRDB v22.2, but the E2E tests
+	// use v21.2.
+	addTransactionMetadataTableQueryWithBasicTTL = `
+		CREATE TABLE transaction_metadata (
+			key UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+			expires_at TIMESTAMPTZ,
+			metadata JSONB
+		) WITH (ttl_expire_after = '1d');
+	`
+
+	addTransactionMetadataTableQuery = `
+		CREATE TABLE transaction_metadata (
+			key UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+			expires_at TIMESTAMPTZ,
+			metadata JSONB
+		) WITH (ttl_expiration_expression = 'expires_at', ttl_job_cron = '@daily');
+	`
+
+	// See: https://www.cockroachlabs.com/docs/stable/changefeed-messages#prevent-changefeeds-from-emitting-row-level-ttl-deletes
+	// for why we set ttl_disable_changefeed_replication = 'true'. This isn't stricly necessary as the Watch API will ignore the
+	// deletions of these metadata rows, but no reason to even have it in the changefeed.
+	// NOTE: This only applies on CRDB v24 and later.
+	addTransactionMetadataTableQueryWithTTLIgnore = `
+		CREATE TABLE transaction_metadata (
+			key UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+			expires_at TIMESTAMPTZ,
+			metadata JSONB
+		) WITH (ttl_expiration_expression = 'expires_at', ttl_job_cron = '@daily', ttl_disable_changefeed_replication = 'true');
+	`
+)
+
+func init() {
+	err := CRDBMigrations.Register("add-transaction-metadata-table", "add-integrity-relationtuple-table", addTransactionMetadataTable, noAtomicMigration)
+	if err != nil {
+		panic("failed to register migration: " + err.Error())
+	}
+}
+
+func addTransactionMetadataTable(ctx context.Context, conn *pgx.Conn) error {
+	row := conn.QueryRow(ctx, "select version()")
+	var version string
+	if err := row.Scan(&version); err != nil {
+		return err
+	}
+
+	if strings.Contains(version, "v22.1") {
+		if _, err := conn.Exec(ctx, addTransactionMetadataTableQueryWithBasicTTL); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if strings.Contains(version, "v24.") {
```

Comment:
Once we move to CRDB v25, this won't work and it will fall back to the behaviour in CRDB 23 and v22.
We need semver comparison logic, ideally, we parse the version and then use the semver library (I assume such library must exist). It could be turn into a small helper over a pgx.Conn.

Reply:
Switched to semver